### PR TITLE
[#12048] Add deep comparison for entities in `verifyEquals` for E2E

### PR DESCRIPTION
**Pull request by domlimm**
Part of #12048.

### CODE REVIEW

File: src/test/java/teammates/test/BaseTestCaseWithSqlDatabaseAccess.java [hunk 1]
```diff
@@ -47,20 +65,113 @@ protected void verifyEquals(BaseEntity expected, ApiOutput actual) {
...
-            assertEquals(expectedQuestionDetails.getQuestionText(), actualQuestion.getQuestionBrief());
             assertEquals(expectedQuestion.getDescription(), actualQuestion.getQuestionDescription());
-            assertEquals(expectedQuestionDetails.getQuestionType(), actualQuestion.getQuestionType());
             assertEquals(expectedQuestion.getGiverType(), actualQuestion.getGiverType());
             assertEquals(expectedQuestion.getRecipientType(), actualQuestion.getRecipientType());
-            // TODO: compare the rest of the attributes D:
+            assertEquals(expectedQuestion.getNumOfEntitiesToGiveFeedbackTo(),
+                    actualQuestion.getCustomNumberOfEntitiesToGiveFeedbackTo());
+            assertEquals(expectedQuestionDetails.getQuestionText(), actualQuestion.getQuestionBrief());
+            assertEquals(expectedQuestionDetails.getQuestionType(), actualQuestion.getQuestionType());
```
Comment (cedricongjh) [inline src/test/java/teammates/test/BaseTestCaseWithSqlDatabaseAccess.java:74]: we should compare the json string of the question details, similar to feedbackresponse

File: src/test/java/teammates/test/BaseTestCaseWithSqlDatabaseAccess.java [hunk 2]
```diff
@@ -47,20 +65,113 @@ protected void verifyEquals(BaseEntity expected, ApiOutput actual) {
...
+        } else if (expected instanceof Student) {
+            Student expectedStudent = (Student) expected;
+            StudentData actualStudent = (StudentData) actual;
+            assertEquals(expectedStudent.getCourseId(), actualStudent.getCourseId());
+            assertEquals(expectedStudent.getName(), actualStudent.getName());
+            assertEquals(expectedStudent.getEmail(), actualStudent.getEmail());
+            assertEquals(expectedStudent.getRegKey(), actualStudent.getKey());
+            assertEquals(expectedStudent.getComments(), actualStudent.getComments());
+            assertEquals(expectedStudent.getTeamName(), actualStudent.getTeamName());
+            assertEquals(expectedStudent.getSectionName(), actualStudent.getSectionName());
```
Comment (cedricongjh) [inline src/test/java/teammates/test/BaseTestCaseWithSqlDatabaseAccess.java:161]: for getTeamName and getSectionName, there might be null pointer exceptions as a student might not have a team or section, let's comment these out and fix this bug in another PR

Comment (domlimm) [reply src/test/java/teammates/test/BaseTestCaseWithSqlDatabaseAccess.java:161]: ah that makes sense. thank you!

**Review by cedricongjh — APPROVED** [review]
LGTM, thanks!

**Pull request merged**